modules/history/src/views/full/index.jsx

```diff
@@ -0,0 +1,342 @@
+
+import classnames from 'classnames'
+
+import { TiRefresh } from 'react-icons/ti'
+import { GoX } from 'react-icons/go'
+import { MdFileDownload } from 'react-icons/md'
+import { FiLink } from 'react-icons/fi'
+import { MdSearch } from 'react-icons/md'
+import { FaFilter } from 'react-icons/fa'
+
+function QueryOptions(props) {
+  return (
+    <div className={style['query-options']}>
+      <div className={style['query-options-daypick']}>
+        <div className={style['query-options-from_to']}>from:</div>
+        <div className={style['daypicker-item']}>
+          <DayPickerInput value={props.defaultFrom} onDayChange={props.handleFromChange} />
+        </div>
+      </div>
+      <div className={style['query-options-daypick']}>
+        <div className={style['query-options-from_to']}>to:</div>
+        <div className={style['daypicker-item']}>
+          <DayPickerInput value={props.defaultTo} onDayChange={props.handleToChange} />
+        </div>
+      </div>
+    </div>
+  )
+}
+
+class ConversationPicker extends React.Component {
+  state = {
+    displayQueryOptions: false
+  }
+
+  toggleFilters() {
+    this.setState({ displayQueryOptions: !this.state.displayQueryOptions })
+  }
+
+  render() {
+    return (
+      <div className={style['conversations']}>
+        <div className={style['conversations-titlebar']}>
+          <div>Conversations</div>
+          <div className={style['conversations-icons']}>
+            <FaFilter className={style['conversations-filter']} onClick={this.toggleFilters.bind(this)} />
+            <TiRefresh className={style['conversations-refresh']} onClick={this.props.refresh} />
+          </div>
+        </div>
+        {this.state.displayQueryOptions && (
+          <QueryOptions
+            handleFromChange={this.props.handleFromChange}
+            handleToChange={this.props.handleToChange}
+            defaultFrom={this.props.defaultFrom}
+            defaultTo={this.props.defaultTo}
+          />
+        )}
+        <div>
+          {this.props.conversations.map(conv => {
+            return (
+              <div
+                className={style['conversations-entry']}
+                onClick={() => this.props.conversationChosenHandler(conv.id)}
+              >
+                <span className={style['conversations-text']} key={conv.id} value={conv.id}>
+                  {`conversation #${conv.id}`}
+                </span>
+                <span className={style['conversations-count']}>({conv.count})</span>
+              </div>
+            )
+          })}
+        </div>
+      </div>
+    )
+  }
+}
+
+class MessagesViewer extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      inspectorIsShown: false,
+      currentlyFocusedMessage: null
+    }
+  }
+
+  getLastMessageDate = messageGroups => {
+    const messages = messageGroups.flatMap(m => m)
+    return new Date(Math.max(...messages.map(m => new Date(m.createdOn))))
```

Comment:
you could use _.maxBy here where iteration logic is done for you https://lodash.com/docs/4.17.11#maxBy